src/common/user-store/preferences-helpers.ts

```diff
@@ -0,0 +1,216 @@
+import moment from "moment-timezone";
+import path from "path";
+import os from "os";
+import { ThemeStore } from "../../renderer/theme.store";
+import { ObservableToggleSet } from "../utils";
+
+export interface KubeconfigSyncEntry extends KubeconfigSyncValue {
+  filePath: string;
+}
+
+export interface KubeconfigSyncValue { }
+
+interface PreferenceDescription<T, R = T> {
+  fromStore(val: T | undefined): R;
+  toStore(val: R): T | undefined;
+}
+
+const httpsProxy: PreferenceDescription<string | undefined> = {
+  fromStore(val) {
+    return val;
+  },
+  toStore(val) {
+    return val || undefined;
+  },
+};
+
+const shell: PreferenceDescription<string | undefined> = {
+  fromStore(val) {
+    return val;
+  },
+  toStore(val) {
+    return val || undefined;
+  },
+};
+
+const colorTheme: PreferenceDescription<string> = {
+  fromStore(val) {
+    return val || ThemeStore.defaultTheme;
+  },
+  toStore(val) {
+    if (!val || val === ThemeStore.defaultTheme) {
+      return undefined;
+    }
+
+    return val;
+  },
+};
+
+const localeTimezone: PreferenceDescription<string> = {
+  fromStore(val) {
+    return val || moment.tz.guess(true) || "UTC";
+  },
+  toStore(val) {
+    if (!val || val === (moment.tz.guess(true) || "UTC")) {
```

Comment:
`|| ""UTC` predicate is always true due to type conversion, did you mean this?

```suggestion
    if (!val || val === (moment.tz.guess(true) || val === "UTC")) {
```

Reply:
I will change this because it confusing to read. But your suggested fix isn't correct either.

Notice the brackets around `moment.tz.guess(true)` and `"UTC"`. Those mean that `val` is compared strict equals to either the guess or `"UTC"` if it couldn't guess.

Reply:
Oh, I misread that, sorry. Thanks for clarification